### ISSUE TITLE
Pass endpoint dependencies to Jinja context

### ISSUE THREAD

**Issue by hasansezertasan** [volfpeter/fasthx#4]
We can not access endpoint dependencies from Jinja templates as of now. What if we change this block:

https://github.com/volfpeter/fasthx/blob/2003d8c58b163873f98d083c67a63f23a1a30799/fasthx/main.py#L141-L142

as this:

```python
        def render(result: Any, *, context: dict[str, Any], request: Request) -> HTMLResponse:
            result.update({"depends": context})
            return self.templates.TemplateResponse(name=template_name, request=request, context=result)
```

Maybe another name for the key, `private` instead of `depends`?

**Comment by volfpeter** [comment]
I'm not sure it can be guaranteed that the mapping returned by the route will always be mutable.

If you need the route's dependencies in addition to its return value for rendering, and you know that the return value is mutable, then I think the best solution (at least for now) is to subclass `Jinja` and override its `__call__()` method to do what you described.

**Comment by hasansezertasan** [comment]
> I'm not sure it can be guaranteed that the mapping returned by the route will always be mutable.

That's a problem 😞

> If you need the route's dependencies in addition to its return value for rendering, and you know that the return value is mutable, then I think the best solution (at least for now) is to subclass `Jinja` and override its `__call__()` method to do what you described.

Well, I'm not actively developing applications with HTMX or Jinja2 for now. So it's not a real deal for me but it probably (most likely) will one day.

What should we do? Try to find a solution, close this, or keep it open to prevent anyone else from opening a similar issue.

**Comment by hasansezertasan** [comment]
One idea...

```python
        def render(result: Any, *, context: dict[str, Any], request: Request) -> HTMLResponse:
            request.state.depends = context
            return self.templates.TemplateResponse(name=template_name, request=request, context=result)
```

This way, we can access the dependencies and the reast from request.state.depends and we can set it to a variable inside jinja template to access it easily.

Info: `request.state` is commonly used for similar purposes, carrying db sessions, user instances, etc.

**Comment by hasansezertasan** [comment]
> `Jinja` could also have one more default implementation, that returns `{**result, "dependencies": route_context}`.

This looks good.

**Comment by volfpeter** [comment]
It should be fixed by #11

**Comment by hasansezertasan** [comment]
> It should be fixed by #11

Well I didn't checked that but closing this, great work 🚀.
